### PR TITLE
Extension Connexion Management Screen

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/screens/connect-extension/connect-extension.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    constructor(props: IThemeProps<ReturnType<typeof stylesProvider>>) {
+        super(props);
+        this.state = {
+            isConnected: WalletConnectClient.isConnected()
+        };
+    }
+
+    public componentDidMount() {
+        this.connectionInterval = setInterval(() => {
```
Comment (krisboit) [inline src/screens/connect-extension/connect-extension.tsx:42]: this is weird, basically we verify every 500ms if we have connection, and if we don't we delete connection details.
There may be cases when the connection could take longer

Comment (krisboit) [reply src/screens/connect-extension/connect-extension.tsx:42]: i think we should not delete connection

File: src/screens/connect-extension/connect-extension.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    private async disconnectExtension() {
+        if (
+            await DialogComponent.confirm(
+                translate('ConnectExtension.disconnect'),
+                translate('ConnectExtension.disconnectInfo')
+            )
+        ) {
+            WalletConnectClient.disconnect();
```
Comment (krisboit) [inline src/screens/connect-extension/connect-extension.tsx:69]: here we should delete connection info

File: src/screens/connect-extension/connect-extension.tsx (hunk 1 above)
Comment (krisboit) [reply src/screens/connect-extension/connect-extension.tsx:42]: also maybe Wallet connect has some events for connect and disconnect events, this way we could refactor this interval

Comment (NovacAndrei) [reply src/screens/connect-extension/connect-extension.tsx:46]: I made some changes and improvements, but I left a TODO with one more issue to solve. I'll continue to search for a fix.

File: src/core/wallet-connect/wallet-connect-client.ts [hunk 3]
```diff
@@ -146,6 +146,13 @@ export const WalletConnectClient = (() => {
         });
     };
 
+    const onDisconnect = (callback: (payload: any) => void) => {
+        walletConnector &&
+            walletConnector.on('disconnect', (error, payload) => {
```
Comment (krisboit) [inline src/core/wallet-connect/wallet-connect-client.ts:151]: you can do something similar to detect connection.
```
walletConnector.on('connect', ...);
```

**Pull request merged**